src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerGeneralApplicationReceived.java

```diff
@@ -16,18 +32,75 @@
 @Component
 public class CaseworkerGeneralApplicationReceived implements CCDConfig<CaseData, State, UserRole> {
-    public static final String CASEWORKER_GENERAL_APPLICATION_RECEIVED = "caseworker-general-application-received";
+    public static final String CASEWORKER_GENERAL_APPLICATION_RECEIVED = "cw-general-application-received";
 
     @Override
     public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
-        new PageBuilder(configBuilder
+        final PageBuilder pageBuilder = addEventConfig(configBuilder);
+
+        final List<CcdPageConfiguration> pages = asList(
+            new GeneralApplicationSelectApplicationType(),
+            new GeneralApplicationUploadDocument()
+        );
+
+        pages.forEach(page -> page.addTo(pageBuilder));
+    }
+
+    public AboutToStartOrSubmitResponse<CaseData, State> aboutToStart(final CaseDetails<CaseData, State> details) {
+
+        log.info("{} about to start callback invoked for Case Id: {}", CASEWORKER_GENERAL_APPLICATION_RECEIVED, details.getId());
+        final CaseData data = details.getData();
+
+        data.setGeneralApplication(GeneralApplication.builder().build());
+
+        return AboutToStartOrSubmitResponse.<CaseData, State>builder()
+            .data(data)
+            .build();
+    }
+
+    public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(final CaseDetails<CaseData, State> details,
+                                                                       final CaseDetails<CaseData, State> beforeDetails) {
+        log.info("{} about to submit callback invoked for Case Id: {}", CASEWORKER_GENERAL_APPLICATION_RECEIVED, details.getId());
+
+        final CaseData data = details.getData();
+        final GeneralApplication generalApplication = data.getGeneralApplication();
+
+        generalApplication.getGeneralApplicationDocument().setDocumentType(DocumentType.GENERAL_APPLICATION);
+
+        data.getDocuments().setDocumentsUploaded(
+            addDocumentToTop(data.getDocuments().getDocumentsUploaded(), generalApplication.getGeneralApplicationDocument())
+        );
+
+        final ListValue<GeneralApplication> generalApplicationListValue = ListValue.<GeneralApplication>builder()
+            .id(UUID.randomUUID().toString())
+            .value(generalApplication)
+            .build();
+
+        if (isNull(data.getGeneralApplications())) {
+            data.setGeneralApplications(singletonList(generalApplicationListValue));
+        } else {
+            data.getGeneralApplications().add(0, generalApplicationListValue);
+        }
+
+        return AboutToStartOrSubmitResponse.<CaseData, State>builder()
+            .data(data)
+            .state(GeneralApplicationReceived)
+            .build();
+    }
+
+    private PageBuilder addEventConfig(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+        return new PageBuilder(configBuilder
             .event(CASEWORKER_GENERAL_APPLICATION_RECEIVED)
```

Comment:
Is there a reason for changing the text on this?
I'm wondering if there will be an issue with name change and existing events for history